app/com/lucidchart/open/cashy/models/FolderModel.scala

```diff
@@ -0,0 +1,80 @@
+package com.lucidchart.open.cashy.models
+
+import com.lucidchart.open.relate.interp._
+import com.lucidchart.open.relate.SqlResult
+import java.util.Date
+import play.api.db._
+import play.api.Play.current
+
+case class Folder(
+  id: Long,
+  bucket: String,
+  key: String,
+  created: Date,
+  hidden: Boolean
+)
+
+object FolderModel extends FolderModel
+class FolderModel {
+  private val folderParser = { row: SqlResult =>
+    Folder(
+      row.long("id"),
+      row.string("bucket"),
+      row.string("key"),
+      row.date("created"),
+      row.bool("hidden")
+    )
+  }
+
+  def findById(folderId: Long): Option[Folder] = {
+    DB.withConnection { implicit connection =>
+      sql"""SELECT `id`, `bucket`, `key`, `created`, `hidden`
+        FROM `folders`
+        WHERE `id` = $folderId""".asSingleOption(folderParser)
+    }
+  }
+
+  def findByKey(bucketName: String, key: String): Option[Folder] = {
+    DB.withConnection { implicit connection =>
+      sql"""SELECT `id`, `bucket`, `key`, `created`, `hidden`
+        FROM `folders`
+        WHERE `key` = $key AND `bucket` = $bucketName""".asSingleOption(folderParser)
+    }
+  }
+
+  def findByKeys(bucketName: String, keys: List[String]): List[Folder] = {
+    if (keys.isEmpty) {
+      Nil
+    } else {
+      DB.withConnection { implicit connection =>
+        sql"""SELECT `id`, `bucket`, `key`, `created`, `hidden`
+          FROM `folders`
+          WHERE `key` IN ($keys) AND `bucket` = $bucketName""".asList(folderParser)
+      }
+    }
+  }
+
+  def hiddenFolders: List[Folder] = {
```

Comment:
This function could potentially return A LOT of records. Is there any way we can filter the records at least a little and return a subset of all hidden folders?